### ISSUE TITLE
`weight` required attribute for `azurerm_traffic_manager_azure_endpoint` even with `Priority` traffic manager profile

### ISSUE THREAD

**Issue by misspellted** [hashicorp/terraform-provider-azurerm#17542]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.1.5

### AzureRM Provider Version

2.95.0

### Affected Resource(s)/Data Source(s)

azurerm_traffic_manager_azure_endpoint

### Terraform Configuration Files

```hcl
resource "azurerm_resource_group" "example" {
  name     = "example-resources"
  location = "West Europe"
}

resource "azurerm_public_ip" "primary" {
  name                = "primary-public-ip"
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
  allocation_method   = "Static"
  domain_name_label   = "primary-public-ip"
}

resource "azurerm_public_ip" "failover" {
  name                = "failover-public-ip"
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
  allocation_method   = "Static"
  domain_name_label   = "failover-public-ip"
}

resource "azurerm_traffic_manager_profile" "example" {
  name                   = "example-profile"
  resource_group_name    = azurerm_resource_group.example.name
  traffic_routing_method = "Priority"

  dns_config {
    relative_name = "example-profile"
    ttl           = 100
  }

  monitor_config {
    protocol                     = "HTTP"
    port                         = 80
    path                         = "/"
    interval_in_seconds          = 30
    timeout_in_seconds           = 9
    tolerated_number_of_failures = 3
  }

  tags = {
    environment = "Production"
  }
}

resource "azurerm_traffic_manager_azure_endpoint" "primary" {
  name               = "primary-endpoint"
  profile_id         = azurerm_traffic_manager_profile.example.id
  priority           = 1
  target_resource_id = azurerm_public_ip.primary.id
}

resource "azurerm_traffic_manager_azure_endpoint" "failover" {
  name               = "failover-endpoint"
  profile_id         = azurerm_traffic_manager_profile.example.id
  priority           = 2
  target_resource_id = azurerm_public_ip.failover.id
}
```

### Debug Output/Panic Output

```shell
╷
│ Error: Missing required argument
│ 
│   on example.tf line 205, in resource "azurerm_traffic_manager_azure_endpoint" "primary":
│  205: resource "azurerm_traffic_manager_azure_endpoint" "primary" {
│ 
│ The argument "weight" is required, but no definition was found.
╵
```

### Expected Behaviour

The `primary` and `failover` endpoints are validated successfully in preparation to be added to the `example` traffic manager profile.

### Actual Behaviour

The `primary` and `failover` endpoints are not validated, unexpectedly failing for missing required `weight` attribute.

### References

#15178

**Comment by misspellted** [comment]
I'm working out of an older code base (thus the 2.95.0 version of the provider, but since the newer resource type is available, I thought to try to use it instead of the deprecated one. The [`traffic_manager_endpoint` 2.95.0 documentation](https://registry.terraform.io/providers/hashicorp/azurerm/2.95.0/docs/resources/traffic_manager_endpoint#weight) has the `weight` attribute as optional, but the [`traffic_manager_azure_endpoint` 2.95.0 documentation](https://registry.terraform.io/providers/hashicorp/azurerm/2.95.0/docs/resources/traffic_manager_azure_endpoint#weight) marks the `weight` attribute as required, persisting to the [latest version](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/traffic_manager_azure_endpoint#weight).

**Comment by sinbai** [comment]
@misspellted thanks for opening this issue here. The fix has been released in the [v3.14.0 of the Terraform Provider](https://github.com/hashicorp/terraform-provider-azurerm/blob/v3.14.0/CHANGELOG.md), could you verify if it works for you? If yes, are you okey if we close this issue?

**Comment by misspellted** [comment]
@sinbai I'll try verifying it later today - have a few TODOs that are a bit higher in priority, as is normal for work :)

**Comment by sinbai** [comment]
@misspellted  thanks for your time and waiting for your feedback:).

**Comment by rcskosir** [comment]
Thanks for taking the time to submit this issue. It looks like this has been resolved as of v3.14.0. As such, I am going to mark this issue as closed. If you run into this again with the latest Terraform and Provider versions, please feel free to [open a new issue](https://github.com/hashicorp/terraform-provider-azurerm/issues/new/choose).
